src/hooks/lib/preview/compounder/useCurvePreviewDepositFallback.ts

```diff
@@ -0,0 +1,57 @@
+import { Address, useContractRead } from "wagmi"
+
+import { VaultType } from "@/lib/types"
+import useCurvePreviewDepositUnderlying from "@/hooks/lib/preview/useCurvePreviewDepositUnderlying"
+import { useVaultContract } from "@/hooks/lib/useVaultContract"
+
+export default function useCurvePreviewDepositFallback({
+  asset,
+  vaultAddress,
+  token,
+  amount,
+  type,
+  slippage,
+  enabled,
+}: {
+  asset: Address
+  vaultAddress: Address
+  token?: Address
+  amount: string
+  type: VaultType
+  slippage: number
+  enabled?: boolean
+}) {
+  const isUnderlyingAsset = token !== asset
+  const preview = useContractRead({
+    ...useVaultContract(vaultAddress),
+    enabled: !!asset && !isUnderlyingAsset && enabled,
+    functionName: "previewDeposit",
+    args: [BigInt(amount)],
+  })
+
+  const previewUnderlying = useCurvePreviewDepositUnderlying({
+    asset,
+    vaultAddress,
+    token,
+    amount,
+    type,
+    slippage,
+    enabled: !!asset && isUnderlyingAsset && enabled,
+  })
+
+  return isUnderlyingAsset
+    ? {
+        ...previewUnderlying,
+        data: {
+          minAmountWei: BigInt(previewUnderlying?.data.minAmountWei).toString(),
+          resultWei: BigInt(previewUnderlying?.data.resultWei).toString(),
+        },
+      }
+    : {
+        ...preview,
+        data: {
+          minAmountWei: (preview.data ?? BigInt(0)).toString(),
```

Comment:
Follow the same pattern as the previous comment, with a fallback value for when it's undefined:

```ts
minAmountWei: preview.data?.toString() ?? "0"
```